### PR TITLE
Fix memory leak and JTL output error

### PR DESCRIPTION
**Pull request by sebastianlorenzo88**

### CODE REVIEW

**Comment by sebastianlorenzo88** [comment]
This look like a Cobertura plugin's issue, it do not support Java 8 properly 

https://github.com/cobertura/cobertura/issues/176

File: http2/src/main/java/com/blazemeter/jmeter/http2/sampler/HTTP2Request.java [hunk 1]
```diff
@@ -1,133 +1,90 @@
...
     public static final String PATH = "HTTP2Request.path"; // $NON-NLS-1$
-    
-    protected static final int MAX_FRAME_DEPTH = JMeterUtils.getPropDefault("httpsampler.max_frame_depth", 5); // $NON-NLS-1$
-    
- 
-    
-    private static Map<String, HTTP2Connection> connectionList;
-    
+
+    private static ThreadLocal<Map<String, HTTP2Connection>> connections = ThreadLocal.withInitial(HashMap::new);
```
Comment (artem-fedorov) [inline http2/src/main/java/com/blazemeter/jmeter/http2/sampler/HTTP2Request.java]: You can replace this initialization with `ThreadLocal.withInitial(() -> new HashMap<>());` for fix Cobertura warnings

**Comment by artem-fedorov** [comment]
Sure, Cobertura plugin doesn't support Java 8 lambdas
But tests are failed:
```
Results :
Tests in error: 
  addPendingResponsesTest(com.blazemeter.jmeter.http2.sampler.HTTP2ConnectionTest): class org.apache.jorphan.logging.Slf4jLogkitLogger overrides final method isInfoEnabled.()Z
  sendTest(com.blazemeter.jmeter.http2.sampler.HTTP2ConnectionTest): Could not initialize class com.blazemeter.jmeter.http2.sampler.HTTP2StreamHandler
  getHTTPFilesTest(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest): class org.apache.jorphan.logging.Slf4jLogkitLogger overrides final method isInfoEnabled.()Z
  getHTTPFilesTest(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest)
  getPortTest(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest): Could not initialize class com.blazemeter.jmeter.http2.sampler.HTTP2Request
  getPortTest(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest)
  sampleMainTest(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest): Could not initialize class com.blazemeter.jmeter.http2.sampler.HTTP2Request
  sampleMainTest(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest)
  setManagersTest(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest): Could not initialize class com.blazemeter.jmeter.http2.sampler.HTTP2Request
  setManagersTest(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest)
  getSendFileAsPostBodyTest(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest): Could not initialize class com.blazemeter.jmeter.http2.sampler.HTTP2Request
  getSendFileAsPostBodyTest(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest)
  sampleTest1(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest): Could not initialize class com.blazemeter.jmeter.http2.sampler.HTTP2Request
  sampleTest1(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest)
  sampleTest2(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest): Could not initialize class com.blazemeter.jmeter.http2.sampler.HTTP2Request
  sampleTest2(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest)
  sampleTest3(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest): Could not initialize class com.blazemeter.jmeter.http2.sampler.HTTP2Request
  sampleTest3(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest)
  createPostContentTest(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest): Could not initialize class com.blazemeter.jmeter.http2.sampler.HTTP2Request
  createPostContentTest(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest)
  getUrlTest(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest): Could not initialize class com.blazemeter.jmeter.http2.sampler.HTTP2Request
  getUrlTest(com.blazemeter.jmeter.http2.sampler.HTTP2RequestTest)
  onDataTest2(com.blazemeter.jmeter.http2.sampler.HTTP2StreamHandlerTest): class org.apache.jorphan.logging.Slf4jLogkitLogger overrides final method isInfoEnabled.()Z
  onPushTest(com.blazemeter.jmeter.http2.sampler.HTTP2StreamHandlerTest): Could not initialize class kg.apc.emulators.EmulatorJmeterEngine
  onDataTest(com.blazemeter.jmeter.http2.sampler.HTTP2StreamHandlerTest): Could not initialize class kg.apc.emulators.EmulatorJmeterEngine
  onHeadersTest(com.blazemeter.jmeter.http2.sampler.HTTP2StreamHandlerTest): Could not initialize class kg.apc.emulators.EmulatorJmeterEngine
  sampleOccurredTest(com.blazemeter.jmeter.http2.sampler.ResultCollectorTest): JMeter requires the checksum of saveservice properties file to continue
  actionPerformedTest(com.blazemeter.jmeter.http2.sampler.ViewResultFullVisualizerTest): Could not initialize class kg.apc.emulators.EmulatorJmeterEngine
  addSubResultsTest(com.blazemeter.jmeter.http2.sampler.ViewResultFullVisualizerTest): Could not initialize class kg.apc.emulators.EmulatorJmeterEngine
  findNodeTest(com.blazemeter.jmeter.http2.sampler.ViewResultFullVisualizerTest): Could not initialize class kg.apc.emulators.EmulatorJmeterEngine
  getResponseAsStringTest(com.blazemeter.jmeter.http2.sampler.ViewResultFullVisualizerTest): Could not initialize class kg.apc.emulators.EmulatorJmeterEngine
Tests run: 38, Failures: 0, Errors: 31, Skipped: 0
```

**Comment by undera** [comment]
"Hope" is not enough, guys :) Can we be sure?

**Comment by sebastianlorenzo88** [comment]
@artem-fedorov That's all the changes for now. 

@undera We are pretty sure that the PR doesn't break anything

**Pull request merged**